src/Executor/CommandLine.cs

```diff
@@ -0,0 +1,1857 @@
+    /// message in this exception was designed to be suitable to print to the user directly.
+    ///
+    /// #CommandLineHelp
+    ///
+    /// The parser also can generate help that is correct for the qualifier and parameter definitions. This can be
+    /// accessed from the code:CommandLineParser.GetHelp method. It is also possible to get the help for just a
+    /// particular Parameter set with code:CommandLineParser.GetHelpForParameterSet. This help includes command
+    /// line syntax, whether the qualifier or parameter is optional or a list, the types of the qualifiers and
+    /// parameters, the help text, and default values.   The help text comes from the 'Define' Methods, and is
+    /// properly word-wrapped.  Newlines in the help text indicate new paragraphs.
+    ///
+    /// #AutomaticExceptionProcessingAndHelp
+    ///
+    /// In the CommandLineParserExample1, while the command line parser did alot of the work there is still work
+    /// needed to make the application user friendly that pretty much all applications need. These include
+    ///
+    ///     * Call the code:CommandLineParser constructor and code:CommandLineParser.CompleteValidation
+    ///     * Catch any code:CommandLineParserException and print a friendly message
+    ///     * Define a -? qualifier and wire it up to print the help.
+    ///
+    /// Since this is stuff that all applications will likely need the
+    /// code:CommandLineParser.ParseForConsoleApplication was created to do all of this for you, thus making it
+    /// super-easy to make a production quality parser (and concentrate on getting your application logic instead
+    /// of command line parsing. Here is an example which defines a 'Ping' command. If you will notice there are
+    /// very few lines of code that are not expressing something very specific to this applications. This is how
+    /// it should be!
+#if EXAMPLE2
+        class CommandLineParserExample2
+        {
+            static void Main()
+            {
+	            // Step 1: Initialize to the defaults
+	            string Host = null;
+	            int Timeout = 1000;
+	            bool Forever = false;
+
+	            // Step 2: Define the parameters, in this case there is only the default parameter set.
+	            CommandLineParser.ParseForConsoleApplication(args, delegate(CommandLineParser parser)
+	            {
+	                parser.DefineOptionalQualifier("Timeout", ref Timeout, "Timeout in milliseconds to wait for each reply.");
+	                parser.DefineOptionalQualifier("Forever", ref Forever, "Ping forever.");
+	                parser.DefineDefaultParameterSet("Ping sends a network request to a host to reply to the message (to check for liveness).");
+	                parser.DefineParameter("Host", ref Host, "The Host to send a ping message to.");
+	            });
+
+	            // Step 3, use the parameters
+	            Console.WriteLine("Got {0} {1} {2} {3}", Host, Timeout, Forever);
```

Comment:
There are 4 placeholders for 3 parameters here.  Also,  this output is only useful if you have the source code sitting next to you so you can see what the values represent.